packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -205,10 +242,13 @@ class ButtonThemeData extends Diagnosticable {
 
   /// Defines a button's base colors, and the defaults for the button's minimum
   /// size, internal padding, and shape.
+  ///
+  /// Despite the name, this property is not a [TextTheme], its value is not a
+  /// collection of [TextStyle]s.
   final ButtonTextTheme textTheme;
 
-  /// Defines whether a button bar should size itself with a minimum size
-  /// constraint or padding.
+  /// Defines whether a [ButtonBar] should size itself with a minimum size
+  /// constraint or with padding.
   ///
   /// Defaults to [ButtonBarLayoutBehavior.padded].
   final ButtonBarLayoutBehavior layoutBehavior;
```

Comment:
Is there a way we could detect an M2 theme and switch to the constrained layout behavior?

Reply:
New themes can specify constrained layout behavior.